core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/workflow/impl/PositiveDoubleProgressStep.java

```diff
@@ -0,0 +1,46 @@
+package uk.ac.bbsrc.tgac.miso.core.data.workflow.impl;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+import uk.ac.bbsrc.tgac.miso.core.data.workflow.AbstractProgressStep;
+import uk.ac.bbsrc.tgac.miso.core.data.workflow.WorkflowStep;
+
+@Entity
+@Table(name = "StepPositiveDouble")
+public class PositiveDoubleProgressStep extends AbstractProgressStep {
+  private static final long serialVersionUID = 1L;
+
+  private double input;
+
+  public double getInput() {
+    return input;
+  }
+
+  public void setInput(double input) {
+    this.input = input;
+  }
+
+  @Override
+  public int hashCode() {
+    final int prime = 31;
+    int result = super.hashCode();
+    result = prime * result + (int) input; // not really sure what's supposed to go here
```

Comment:
This is fine, since `input` is the only attribute of this class nothing else needs to go here.